lib/src/linting/rules/no-nullable-fields-within-requests.spec.ts

```diff
@@ -0,0 +1,207 @@
+import { HttpMethod } from "../../models/http";
+import {
+  ApiNode,
+  BodyNode,
+  EndpointNode,
+  RequestNode
+} from "../../models/nodes";
+import {
+  INT32,
+  NULL,
+  objectType,
+  referenceType,
+  STRING,
+  TypeKind,
+  unionType
+} from "../../models/types";
+import { fakeLocatable } from "../../spec-helpers/fake-locatable";
+import { noNullableFieldsWithinRequests } from "./no-nullable-fields-within-requests";
+
+describe("rule: no nullable fields within request body", () => {
+  test("valid for correct usage", () => {
+    const errors = noNullableFieldsWithinRequests({
+      api: fakeLocatable<ApiNode>({
+        name: fakeLocatable("example-api")
+      }),
+      endpoints: [
+        // Endpoint with response payload
+        fakeLocatable<EndpointNode>({
+          name: fakeLocatable("listUsers"),
+          method: fakeLocatable<HttpMethod>("GET"),
+          path: fakeLocatable("/users"),
+          isDraft: false,
+          tests: [],
+          request: fakeLocatable<RequestNode>({
+            body: fakeLocatable<BodyNode>({
+              type: {
+                kind: TypeKind.STRING
+              }
+            })
+          }),
+          responses: []
+        }),
+        // Endpoint with reference type in response payload
+        fakeLocatable<EndpointNode>({
+          name: fakeLocatable("createUser"),
+          method: fakeLocatable<HttpMethod>("POST"),
+          path: fakeLocatable("/users"),
+          isDraft: false,
+          tests: [],
+          request: fakeLocatable<RequestNode>({
+            body: fakeLocatable<BodyNode>({
+              type: objectType([
+                {
+                  name: "data",
+                  type: objectType([
+                    {
+                      name: "slug",
+                      type: STRING,
+                      optional: true
+                    }
+                  ]),
+                  optional: false
+                }
+              ])
+            })
+          }),
+          responses: []
+        })
+      ],
+      types: [
+        {
+          name: "responseBody201",
```

Comment:
did you mean to make the response nullable instead of optional?

Reply:
my bad - this is confusing. That's actually the request body i'm describing here.